### PR TITLE
FIX: Only pass connection to sqlalchemy engine in JdbcHook

### LINKED ISSUE #42664: In MsSqlHook, SQLAlchemy engine scheme is overriden by the change in #40669
### Apache Airflow Provider(s)

common-sql

### Versions of Apache Airflow Providers

apache-airflow-providers-common-sql==1.16.0
apache-airflow-providers-microsoft-mssql==3.9.0
apache-airflow-providers-odbc==4.7.0

### Apache Airflow version

2.10.2

### Operating System

RHEL 8.10

### Deployment

Virtualenv installation

### Deployment details

_No response_

### What happened

We're using mssql+pyodbc connections in airflow to connect to our MS SQL Server. We're creating a hook using `MsSqlHook` and using `.get_sqlalchemy_engine()` to get an sql alchemy engine object to pass along our database tasks.
Prior to apache-airflow-providers-common-sql 1.15.0, this was working as expected (last known working version is 1.14.1, please see below for a working example)
With a change introduced in 1.15.0 (https://github.com/apache/airflow/pull/40669), `.get_sqlalchemy_engine()` passes the `creator` argument with the value `self.get_conn` to the `.create_engine()` function, which overrides the engine creation with the default connection scheme, which uses `pymssql`, even though we specifically define the scheme as `pyodbc` in the connection, and in the hook.

Resulting engine object becomes a weird combination of two, with a `mssql+pyodbc` scheme in the connection uri, but tries to connect to using the `pymssql` dialect internally, which results in fatal errors:

image

As if it's trying to make a `pymssql` connection, but with a `pyodbc` URL.

### What you think should happen instead

The connection should work, as it does prior to the change.

image

If we comment out / delete the line `engine_kwargs["creator"] = self.get_conn` in `‎airflow/providers/common/sql/hooks/sql.py` (https://github.com/apache/airflow/commit/f6c7388cfa70874d84f312a5859a4f510fef0084#diff-6e1b2f961cb951d05d66d2d814ef5f6d8f8bf8f43c40fb5d40e27a031fed8dd7R246), connections works as expected.

### How to reproduce

This is the connection that's used (values changed for privacy)

Basically use any MSSQL connection with a `mssql+pyodbc` scheme

```
{
    "conn_type": "mssql",
    "host": "<redacted>",
    "login": "user",
    "password": "pass",
    "schema": "master",
    "port": 1433,
    "extra": {
        "driver": "ODBC Driver 18 for SQL Server",
        "encrypt": "yes",
        "sqlalchemy_scheme": "mssql+pyodbc"
    }
}
```

Try to connect through a hook and sqlalchemy engine:

```python
from airflow.providers.microsoft.mssql.hooks.mssql import MsSqlHook
hook = MsSqlHook(mssql_conn_id='conn1')  # conn1: the id for above connection
#hook = MsSqlHook(mssql_conn_id='conn1', sqlalchemy_scheme='mssql+pyodbc') # optional, same result
engine = hook.get_sqlalchemy_engine()
engine.connect()
```

This fails on > 1.15.0
Works on <= 1.14.1

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by dabla**
As mentioned in issue [42644](https://github.com/apache/airflow/issues/42664), the connection should only be passed to create the sqlalchemy engine for the JdbcHook, and not in a generic way in the DbApiHook, so that mixed usage of sqlalchemy engine and database connections is still possible (e.g. odbc with pymssql for example).

closes: #42664

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/common/sql/hooks/sql.py [hunk 1]
```diff
@@ -272,10 +272,6 @@ def get_sqlalchemy_engine(self, engine_kwargs=None):
         :param engine_kwargs: Kwargs used in :func:`~sqlalchemy.create_engine`.
         :return: the created engine.
         """
-        if engine_kwargs is None:
```
Comment (emredjan) [inline airflow/providers/common/sql/hooks/sql.py:275]: I think this part should still stay? Only line 277 needs to be removed.

Comment (dabla) [reply airflow/providers/common/sql/hooks/sql.py:275]: done

File: airflow/providers/common/sql/hooks/sql.py [hunk 2]
```diff
@@ -267,14 +267,13 @@ def sqlalchemy_url(self) -> URL:
 
     def get_sqlalchemy_engine(self, engine_kwargs=None):
         """
-        Get an sqlalchemy_engine object.
```
Comment (shahar1) [inline airflow/providers/common/sql/hooks/sql.py]: Actually this is correct
https://oneminuteenglish.org/an-s-or-a-s-which-is-correct/

File: airflow/providers/jdbc/hooks/jdbc.py [hunk 3]
```diff
@@ -163,6 +163,19 @@ def sqlalchemy_url(self) -> URL:
             database=conn.schema,
         )
 
+    def get_sqlalchemy_engine(self, engine_kwargs=None):
```
Comment (shahar1) [inline airflow/providers/jdbc/hooks/jdbc.py:166]: Could you please add some tests?

File: airflow/providers/common/sql/hooks/sql.py (hunk 2 above)
Comment (dabla) [reply airflow/providers/common/sql/hooks/sql.py]: reverted

File: airflow/providers/jdbc/hooks/jdbc.py (hunk 3 above)
Comment (dabla) [reply airflow/providers/jdbc/hooks/jdbc.py:166]: will do

**Review by shahar1 — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**